build-tools/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt

```diff
@@ -311,3 +320,233 @@ private fun sshExecutor(project: Project) : ExecutorService = object : ExecutorS
+                        debugger.HandleCommand("exit %d" % code)
+                    
+                    def __lldb_init_module(debugger, _):
+                        debugger.HandleCommand('command script add -f lldb_cmd.exit_code get_exit_code')
+                """.trimIndent())
+        absolutePath
+    }
+
+    private fun kill() = project.exec {
+        it.commandLine(idb, "kill")
+    }
+
+    private fun targetUDID(): String {
+        val out = ByteArrayOutputStream()
+        // idb launches idb_companion but doesn't wait for it and just exits.
+        // So relaunch `list-targets` again.
+        for (i in 1..3) {
+            project.exec {
+                it.commandLine(idb, "list-targets", "--json")
+                it.standardOutput = out
+            }.assertNormalExitValue()
+            if (out.toString().trim().isNotEmpty()) break
+            else TimeUnit.SECONDS.sleep(i.toLong())
+        }
+        return out.toString().run {
+            check(isNotEmpty())
+            @Serializable
+            data class DeviceTarget(val name: String, val udid: String, val state: String, val type: String)
+            split("\n")
+                    .filter { it.isNotEmpty() }
+                    .map { Json(strictMode = false).parse(DeviceTarget.serializer(), it) }
+                    .first {
+                        it.type == "device" && deviceName?.run { this == it.name } ?: true
+                    }
+                    .udid
+        }
+    }
+
+    private fun install(udid: String, bundlePath: String) {
+        val out = ByteArrayOutputStream()
+
+        val result = project.exec {
+            it.workingDir = xcProject.toFile()
+            it.commandLine = listOf(idb, "install", "--udid", udid, bundlePath)
+            it.standardOutput = out
+            it.errorOutput = out
+            it.isIgnoreExitValue = true
+        }
+        println(out.toString())
+        check(result.exitValue == 0) { "Installation of $bundlePath failed: $out" }
+    }
+
+    private fun uninstall(udid: String, bundleId: String) {
+        val out = ByteArrayOutputStream()
+
+        project.exec {
+            it.workingDir = xcProject.toFile()
+            it.commandLine = listOf(idb, "uninstall", "--udid", udid, bundleId)
+            it.standardOutput = out
+            it.errorOutput = out
+            it.isIgnoreExitValue = true
+        }
+        println(out.toString())
+    }
+
+    private fun startDebugServer(udid: String, bundleId: String): String {
+        val out = ByteArrayOutputStream()
+
+        val result = project.exec {
+            it.workingDir = xcProject.toFile()
+            it.commandLine = listOf(idb, "debugserver", "start", "--udid", udid, bundleId)
+            it.standardOutput = out
+            it.errorOutput = out
+        }
+        check(result.exitValue == 0) { "Failed to start debug server: $out" }
+        return out.toString()
+    }
+}
+
+fun KonanTestExecutable.xcodeBuild() {
```

Comment:
IMHO it may worth renaming this method to something like `configureXcodeBuild`. Just `xcodeBuild` may make an impression that the build is preformed right now while it will be performed only before task execution.